sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1659,11 +1665,12 @@ object TimeWindowing extends Rule[LogicalPlan] {
           val windowEnd = windowStart + window.windowDuration
 
           CreateNamedStruct(
-            Literal(WINDOW_START) :: windowStart ::
```

Comment:
Previously we manually set the output of Expand here as `TimestampType` (`windowAttr`). As `windowStart` and `windowEnd` are producing long values, when we infer output from Expand's projections, we will get `LongType` instead of `TimestampType`. So we need to explicitly convert the `LongType` to `TimestampType`.